activerecord/test/cases/cache_key_test.rb

```diff
@@ -0,0 +1,25 @@
+require "cases/helper"
+
+module ActiveRecord
+  class CacheKeyTest < ActiveRecord::TestCase
+    self.use_transactional_tests = false
+
+    class CacheMe < ActiveRecord::Base; end
+
+    setup do
+      @connection = ActiveRecord::Base.connection
+      @connection.create_table(:cache_mes) { |t| t.timestamps }
+    end
+
+    teardown do
+      @connection.drop_table :cache_mes, if_exists: true
+    end
+
+    test "test_cache_key_format_is_not_too_precise" do
+      record = CacheMe.create
+      key = record.cache_key
+
+      assert_equal key, record.reload.cache_key
+    end
+  end
+end
```

Comment:
Yes. I experimented with that, but got stuck on the following:
- I couldn't define the model inline
- I needed to turn off transactional testing otherwise the MySQL test suite kept failing

I thought it would be cleaner if I put that setup in a separate file instead of dirtying the existing one. Would you prefer it if I moved it all back to `integration_test.rb`?